src/Models/Traits/ValidatesExpectedTotal.php

```
@@ -0,0 +1,93 @@
+        if ($this->shouldThrowOnExpectedTotalMismatch()) {
+            $errorMessage .= sprintf(': expected %.2f, got %.2f (difference: %.2f)', $expectedTotal, $actualTotal, abs($expectedTotal - $actualTotal));
+
+            // determine which exception to throw based on the class type
+            if ($isInvoice) {
+                throw new InvalidInvoiceException($errorMessage);
+            } elseif ($isUsageReceipt) {
+                throw new InvalidUsageReceiptException($errorMessage);
+            } else {
+                // fallback: should not happen, but log and throw generic exception
+                throw new InvalidInvoiceException($errorMessage);
+            }
```

PHPMD (2.15.0)

The method validateExpectedTotal uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)